tensorflow/go/operation.go

```diff
@@ -123,6 +143,63 @@ func (p Output) c() C.TF_Output {
 
 func (p Output) canBeAnInput() {}
 
+// Consumers returns the inputs that consume this output.
+func (p Output) Consumers() []Consumer {
+	max := int(C.TF_OperationOutputNumConsumers(p.c()))
+	inputs := make([]C.TF_Input, max)
+	n := C.TF_OperationOutputConsumers(p.c(), (*C.TF_Input)(unsafe.Pointer(&inputs[0])), C.int(max))
+	inputs = inputs[:int(n)]
+
+	var consumers []Consumer
+	for _, consumer := range inputs {
+		consumers = append(consumers, Consumer{
+			Index: int(consumer.index),
+			Op: &Operation{
+				c: consumer.oper,
+				g: p.Op.g,
+			},
+		})
+	}
+
+	return consumers
+}
+
+// Consumer is one of the inputs to an operation. It consumes an output from an
+// operation.
+type Consumer struct {
+	// Op is the Operation that corresponds to this consumer.
+	Op *Operation
+
+	// Index specifies the index of the input within the Operation.
+	Index int
+}
+
+func (p Consumer) c() C.TF_Input {
+	if p.Op == nil {
+		// Attempt to provide a more useful panic message than "nil
+		// pointer dereference".
+		panic("nil-Operation. This Consumer object was not created by a call to Output.Consumers or Operation.Input")
```

Comment:
Update the error message (since there is no `Operation.Input`). How about:

```go
panic("nil-Operation. Consumer objects should only be created by a call to Output.Consumers")
```

or something like that.

Reply:
Done